push/src/curl_wrapper.cc

```diff
@@ -40,7 +46,26 @@ int CurlWrapper::performHttpRequest(HttpMethod method, const std::string& uri,
   curl_easy_setopt(curl_, CURLOPT_URL, uri.c_str());
 
   curl_slist* header_chunk = nullptr;
-  header_chunk = curl_slist_append(header_chunk, CONTENT_TYPE);
+  header_chunk = curl_slist_append(header_chunk, CONTENT_TYPE); 
+  if (nullptr == header_chunk)
+  {
+    return -1;
+  }
+
+  curl_slist* optHeader_tmp = optHttpHeader_;
```

Comment:
Is there a reason for cloning the header list? If you remove the deletion of the working list in line 105 (`curl_slist_free_all(header_chunk);`) you should be able to re-use the one stored in the member variable.

Reply:
Yes, optHttpHeader_ need be re-use. So I cloning the header list in line 56 (curl_slist* header_tmp = curl_slist_append(header_chunk, optHeader_tmp->data);)